acme/acme/client.py

```diff
@@ -624,7 +627,31 @@ def _send_request(self, method, url, *args, **kwargs):
         kwargs.setdefault('headers', {})
         kwargs['headers'].setdefault('User-Agent', self.user_agent)
         kwargs.setdefault('timeout', self._default_timeout)
-        response = self.session.request(method, url, *args, **kwargs)
+        try:
+            response = self.session.request(method, url, *args, **kwargs)
+        except requests.exceptions.RequestException as e:
+            #pylint: disable=pointless-string-statement
+            """Requests response parsing
+
+            The requests library emits exceptions with a lot of extra text.
+            We parse them with a regexp to raise a more readable exceptions.
+
+            Example:
+            HTTPSConnectionPool(host='acme-v01.api.letsencrypt.org',
+            port=443): Max retries exceeded with url: /directory
+            (Caused by NewConnectionError('
+            <requests.packages.urllib3.connection.VerifiedHTTPSConnection
+            object at 0x108356c50>: Failed to establish a new connection:
+            [Errno 65] No route to host',))"""
+
+            err_regex = r".*host='(\S*)'.*url\: (\/\w*).*(\[Errno \d+\])([A-Za-z ]*)"
```

Comment:
Ah, I probably wasn't clear enough. It's not important to me that you have a `^` immediately before HTTPSConnectionPool, just that the string be in the regex. That is, I originally proposed:

`err_regex = r"^HTTPSConnectionPool\(host='(\S*)`

But based on the actual message string you postedrecently, it should be:

`err_regex = r"HTTPSConnectionPool\(host='(\S*)`

(that is, remove the `^`). I think that shouldn't require changing exception handling or tests. If that's incorrect, please let me know what it would require changing. Thanks!